### ISSUE TITLE
NPE when creating a Lucine index

### ISSUE THREAD

**Issue by antonzherdev** [orientechnologies/orientdb#6703]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .9[X] .rc1[ ] .beta2[ ] .beta1[ ]
## Operating System
- [X] Linux
- [X] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 
## Expected behavior and actual behaviour

Null pointer exceptions when creating a Lucine index in the log:

```
ERROR c.o.o.c.s.i.l.p.OLocalPaginatedStorage [lt-dispatcher-9] - $ANSI{green {db=wdb}} Error while flushing index via index engine of class OLuceneIndexEngineDelegate.
java.lang.NullPointerException: null
        at com.orientechnologies.lucene.engine.OLuceneIndexEngineDelegate.flush(OLuceneIndexEngineDelegate.java:69)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.synch(OAbstractPaginatedStorage.java:2428)
        at com.orientechnologies.orient.core.index.OIndexManagerShared.createIndex(OIndexManagerShared.java:197)
        at com.orientechnologies.orient.core.index.OIndexManagerProxy.createIndex(OIndexManagerProxy.java:87)
        at com.erp.db.WorkspaceSearch.check(WorkspaceSearch.scala:84)
```

Nevertheless, the index is working fine after this. 
Probably there should be simply a null check in OLuceneIndexEngineDelegate.flush, because it's there in few other methods.
## Steps to reproduce the problem

```
val itp = OClass.INDEX_TYPE.FULLTEXT.toString
val factory: OIndexFactory = OIndexes.getFactory(itp, null)
db.getMetadata.getIndexManager.createIndex(name, itp,
        new OSimpleKeyIndexDefinition(factory.getLastVersion, OType.STRING), null, null, null,
        OLuceneIndexFactory.LUCENE_ALGORITHM)
```

**Comment by robfrank** [comment]
I wasn't able to repoduce the problem with your code (Scala, I guess).
But, as i understand, your suggestion is to check for null on the flush method:
```
@Override
  public void flush() {
    delegate.flush();
  }
```
to
```
@Override
  public void flush() {
    if (delegate != null)
    delegate.flush();
  }
```

I will check the lifecycle before add the line;  I don't like defensive programming

**Comment by robfrank** [comment]
I was unable to replicate the problem with the code snippet provided (is it Scala?) using either 2.2.9 and 2.2.10

**Comment by antonzherdev** [comment]
@robfrank I found that it's not always reproducible. If I create a fresh database it's working fine but with the database which I have currently it always gives the exception. 

I do not think that ` if (delegate != null)` is a good solution but there few similar checks in the class already. I guess it should never happen and the exception means that something can be wrong with the index lifecycle. It can happen only if flush is called before init.

**Comment by robfrank** [comment]
yes, you're right. We will investigate on the lifecycle.

**Comment by robfrank** [comment]
Hi, we reviewed the life cycle, dropped the delegate/delegator and improved other parts of lucene support. All this fixes will be in the next 2.2.14 release. Fell free to reopen the issue if something goes wrong.
